tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs

```diff
@@ -65,7 +76,55 @@ type ErrorResponse =
             Error 10, Line 7, Col 12, Line 7, Col 14, "Unexpected symbol '|}' in field declaration. Expected identifier or other token."
             Error 10, Line 10, Col 17, Line 10, Col 21, "Incomplete structured construct at or before this point in field declaration. Expected identifier or other token."
         ]
+        
+    [<Fact>]
+    let ``Anonymous Record type annotation with with fields defined in a record`` () =
+        Fsx """
+type T = { ff : int }
 
+let t3 (t1: {| gu: string; ff: int |}) = { t1 with ff = 3 }
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3578, Line 4, Col 52, Line 4, Col 54, "'ff' is part of an anonymous record. Use {| expr with ff = ... |} instead.")
+        ]
+        
+    [<Fact>]
+    let ``This expression was expected to have an anonymous Record but has a record`` () =
+        Fsx """
+let t3 (t1: {| gu: string; ff: int |}) = { t1 with ff = 3 }
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3578, Line 2, Col 52, Line 2, Col 54, "'ff' is part of an anonymous record. Use {| expr with ff = ... |} instead.")
+        ]
+        
+    [<Fact>]
+    let ``This expression was expected to have an struct anonymous Record but has a record`` () =
+        Fsx """
+let t3 (t1: struct {| gu: string; ff: int |}) = { t1 with ff = 3 }
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3578, Line 2, Col 59, Line 2, Col 61, "'ff' is part of an anonymous record. Use {| expr with ff = ... |} instead.")
+        ]
+        
+    [<Fact>]
+    let ``This expression was expected to have an anonymous with various properties Record but has a record`` () =
+        Fsx """
+let f (r: {| A: int; C: int |}) = { r with A = 1; B = 2; C = 3 }
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3578, Line 2, Col 44, Line 2, Col 45, "'A' is part of an anonymous record. Use {| expr with A = ... |} instead.")
+            (Error 39, Line 2, Col 51, Line 2, Col 52, "The record label 'B' is not defined.")
+            (Error 3578, Line 2, Col 58, Line 2, Col 59, "'C' is part of an anonymous record. Use {| expr with C = ... |} instead.")
```

Comment:
Would it make sense to do something like this? Underline the braces (for which we probably have to output 2 diagnostics) and not talk about fields that are correct?

```suggestion
            (Error 3578, Line 2, Col 35, Line 2, Col 35, "This expression is an anonymous record, use {|...|} instead of {...}.")
            (Error 39, Line 2, Col 51, Line 2, Col 52, "The record label 'B' is not defined.")
            (Error 3578, Line 2, Col 64, Line 2, Col 65, "This expression is an anonymous record, use {|...|} instead of {...}.")
```

Reply:
I think so. Let me check if I have the range available

Reply:
@0101 Ok took me some time but my last commit might do what was suggested. Please take a look and let me know if that is what you prefer and if the ranges are where it is required :)

Reply:
Looks good! I guess we lost the diagnostic for invalid field (B), but I'm fine with that.